### PR TITLE
fix(AAP-30273): allow updating is_superuser field when local resource management is disallowed

### PR DESCRIPTION
**Pull request by Dostonbek1**
JIRA: [AAP-30273](https://issues.redhat.com/browse/AAP-30273)

### CODE REVIEW

File: src/aap_eda/api/serializers/mixins.py [hunk 1]
```diff
@@ -15,13 +15,22 @@
...
+from aap_eda.core import models
 
 
 class SharedResourceSerializerMixin:
-    def validate_shared_resource(self):
+    def validate_shared_resource(self, data=None):
         if not settings.ALLOW_LOCAL_RESOURCE_MANAGEMENT:
             view = self.context.get("view")
-            action = view.action.capitalize() if view else "Action"
+            action = view.action.capitalize() if view else "action"
```
Comment (Alex-Izquierdo) [inline src/aap_eda/api/serializers/mixins.py]: Should it be "Action"? (capitalized)

**Review by AlanCoding — CHANGES_REQUESTED** [review]
By inheriting from `SharedResourceSerializerMixin`, as opposed to its child class `UserUpdateSerializerBase`, it looks like you've missed out on `validate_is_superuser` which is necessary for permission checking.

The test `test_modify_superuser_as_org_admin` has nice-looking coverage for exactly this, but since the permission check is performed in the serializer, swapping out the serializer can leave this edit unprotected from access rules. So it would probably be a good idea to test that it still gives a 403 with a non-admin client fixture.

**Comment by h-kataria** [comment]
@elyezer please review

File: src/aap_eda/api/views/user.py [hunk 2]
```diff
@@ -196,9 +196,10 @@ def perform_create(self, serializer):
         },
     ),
     partial_update=extend_schema(
-        exclude=not settings.ALLOW_LOCAL_RESOURCE_MANAGEMENT,
         description="Partial update of a user.",
-        request=serializers.UserCreateUpdateSerializer,
+        request=serializers.UserCreateUpdateSerializer
```
Comment (Alex-Izquierdo) [inline src/aap_eda/api/views/user.py:200]: I would prefer a conventional if-else block for readability.

**Review by Alex-Izquierdo — APPROVED** [review]

File: src/aap_eda/api/views/user.py (hunk 2 above)
Comment (Dostonbek1) [reply src/aap_eda/api/views/user.py:200]: already changed. Thanks for catching it.

**Review by AlanCoding — APPROVED** [review]
:ship:

**Review by elyezer — COMMENTED** [review]
From an initial look this doesn't seem that it may cause installation problems. If this does not change behavior when the `ALLOW_LOCAL_RESOURCE_MANAGEMENT` is set to `False` then we should be good to merge.

**Review by Alex-Izquierdo — APPROVED** [review]

**Pull request merged**